src/irr/asset/CGLIWriter.cpp

```diff
@@ -0,0 +1,327 @@
+/*
+MIT License
+Copyright (c) 2019 AnastaZIuk
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+#include "CGLIWriter.h"
+
+#ifdef _IRR_COMPILE_WITH_GLI_WRITER_
+
+#ifdef _IRR_COMPILE_WITH_GLI_
+#include "gli/gli.hpp"
+#else
+#error "It requires GLI library"
+#endif
+
+namespace irr
+{
+	namespace asset
+	{
+		inline std::pair<gli::texture::format_type, std::array<gli::gl::swizzle, 4>> getTranslatedIRRFormat(const IImageView<ICPUImage>::SCreationParams& params);
+
+		template<typename aType>
+		aType getSingleChannel(const void* data)
+		{
+			return *(reinterpret_cast<const aType*>(data));
+		}
+
+		bool CGLIWriter::writeAsset(io::IWriteFile* _file, const SAssetWriteParams& _params, IAssetWriterOverride* _override)
+		{
+			if (!_override)
+				getDefaultOverride(_override);
+
+			SAssetWriteContext ctx{ _params, _file };
+
+			const asset::ICPUImageView* imageView = IAsset::castDown<ICPUImageView>(_params.rootAsset);
+
+			io::IWriteFile* file = _override->getOutputFile(_file, ctx, { imageView, 0u });
+
+			if (!file)
+				return false;
+
+			return writeGLIFile(file, imageView);
+		}
+
+		bool CGLIWriter::writeGLIFile(io::IWriteFile* file, const asset::ICPUImageView* imageView)
+		{
+			os::Printer::log("WRITING GLI: writing the file", file->getFileName().c_str(), ELL_INFORMATION);
+
+			const auto& imageViewInfo = imageView->getCreationParameters();
+			const auto& imageInfo = imageViewInfo.image->getCreationParameters();
+			const auto& image = imageViewInfo.image;
+
+			if (image->getRegions().length() == 0)
+			{
+				os::Printer::log("WRITING GLI: there is a lack of regions!", file->getFileName().c_str(), ELL_INFORMATION);
+				return false;
+			}
+			if (isBlockCompressionFormat(imageInfo.format))
```

Comment:
really? why?

Reply:
so do we want to support however?

Reply:
obviously yes, thats why we had DDS loader in the first place!

Reply:
to do https://github.com/buildaworldnet/IrrlichtBAW/pull/411#discussion_r367044582